drivers/can/can_common.c

```diff
@@ -88,10 +88,8 @@ static int can_calc_timing_int(uint32_t core_clock, struct can_timing *res,
 	int sp_err;
 	struct can_timing tmp_res;
 
-	if (bitrate == 0 || sp >= 1000 ||
-	    (!IS_ENABLED(CONFIG_CAN_FD_MODE) && bitrate > 1000000) ||
-	     (IS_ENABLED(CONFIG_CAN_FD_MODE) && bitrate > 8000000)) {
-		return -EINVAL;
+	if (bitrate == 0 || sp >= 1000) {
+		return -EDOM;
```

Comment:
-EDOM seems reasonably descriptive for bitrates and samples points which are too large specifically, but not as much for the bitrate being zero. This probably doesn't matter given the return contract it fulfills.

Reply:
How come? A bitrate of 0 is outside of the allowed value domain.

Reply:
Yes, I am suggesting to use -EINVAL.